src/storage/write_ahead_log.cpp

```diff
@@ -192,6 +263,11 @@ void WriteAheadLog::WriteCheckpoint(MetaBlockPointer meta_block) {
 	serializer.End();
 }
 
+bool WriteAheadLog::IsEncrypted() const {
+	const auto &config = DBConfig::GetConfig(database.GetDatabase());
+	return config.options.wal_encryption;
+}
+
```

Comment:
Should this setting not persisted in the WriteAheadLog object ? (and restored on load)

I would basically think it's a property set on construction of WriteAheadLog of whether it will be encrypted, without relying on the global setting that otherwise is indeed dangerous.